### PR TITLE
add py-trio and dependencies

### PR DESCRIPTION
**Pull request by jandemter**
#### Description
Trio is "a friendly Python library for async concurrency and I/O".

py-sniffio, py-outcome are the main dependencies,
py-trustme is needed to run the tests for py-trio,
py-contextvars and py-immutables for compatibility with Python 3.6

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.14.6 18G95
Xcode 10.3 10G8

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: python/py-contextvars/Portfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+name                py-contextvars
+version             2.4
+categories-append   devel
+platforms           darwin
+license             Apache-2
+
+python.versions     36
+
+maintainers         @jandemter openmaintainer
```
Comment (reneeotten) [inline python/py-contextvars/Portfile]: please also add an e-mail address (see the [guide](https://guide.macports.org/#development.creating-portfile)), so something like:
```{example.org:jandemter} @jandemter} openmaintainer```

File: python/py-contextvars/Portfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+checksums           rmd160  8a93b02e1ca00c938a16ca579ac32801dd4e4cfb \
+                    sha256  f38c908aaa59c14335eeea12abea5f443646216c4e29380d7bf34d2018e2c39e \
+                    size    9570
+
+if {${name} ne ${subport}} {
+    depends_build-append    port:py${python.version}-setuptools
+    depends_lib-append      port:py${python.version}-immutables
+    depends_test-append     port:py${python.version}-pytest
+
+    set pyver [join [split ${python.version} {}] {.}]
```
Comment (reneeotten) [inline python/py-contextvars/Portfile]: this shouldn't be necessary, to get the correct ```py.test``` version below, you can use the variable ```${python.branch}``` that is defined in the Python PortGroup as in ```test.cmd        py.test-${python.branch}```

File: python/py-contextvars/Portfile [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    test.run            yes
+    test.cmd            py.test-${pyver}
+    test.target
+    test.env            PYTHONPATH=${worksrcpath}/
```
Comment (reneeotten) [inline python/py-contextvars/Portfile]: this typically refers to ```PYTHONPATH=${worksrcpath}/build/lib```

File: python/py-contextvars/Portfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+    livecheck.type          none
+} else {
+    livecheck.type          pypi
```
Comment (reneeotten) [inline python/py-contextvars/Portfile]: this is the default livecheck if you use ```pypi``` in the ```master_sites```, so it's not needed. Please remove the ```else``` clause.

File: python/py-contextvars/Portfile [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-contextvars
+version             2.4
+categories-append   devel
+platforms           darwin
```
Comment (reneeotten) [inline python/py-contextvars/Portfile]: if no architecture-dependent files are installed (I didn't check), then please also add ```supported_archs     noarch```

File: python/py-outcome/Portfile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+PortSystem          1.0
+PortGroup           python 1.0
+
+name                py-outcome
+version             1.0.0
+categories-append   devel
+platforms           darwin
+license             {Apache-2 MIT}
+
+python.versions     27 34 35 36 37
```
Comment (reneeotten) [inline python/py-outcome/Portfile]: please don't add EOL Python versions (i.e., 34) for new ports

File: python/py-trio/Portfile [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    depends_test-append     port:py${python.version}-astor \
+                            port:py${python.version}-openssl \
+                            port:py${python.version}-yapf \
+                            port:py${python.version}-trustme \
+                            port:py${python.version}-jedi \
+                            port:py${python.version}-pylint \
+                            port:py${python.version}-pytest
+
+    if {${python.version} < 37} {
+        depends_lib-append    port:py${python.version}-contextvars
```
Comment (reneeotten) [inline python/py-trio/Portfile]: this doesn't follow the "4 spaces" guideline, please align as above

File: python/py-trustme/Portfile [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    depends_lib-append      port:py${python.version}-cryptography \
+                            port:py${python.version}-idna
+    depends_test-append     port:py${python.version}-service_identity \
+                            port:py${python.version}-pytest \
+                            port:py${python.version}-openssl \
+
+
+    if {${python.version} < 33} {
+        depends_lib-append    port:py${python.version}-ipaddress
+        depends_lib-append    port:py${python.version}-futures
```
Comment (reneeotten) [inline python/py-trustme/Portfile]: this doesn't follow the "4 spaces" guideline, please align as above

File: python/py-contextvars/Portfile [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+supported_archs     noarch
+license             Apache-2
+
+python.versions     36
+
+maintainers         @jandemter {demter.de:jan} openmaintainer
```
Comment (mf2k) [inline python/py-contextvars/Portfile]: This is not correct format. The brackets need to be around your email and GitHub handle.

File: python/py-immutables/Portfile [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+name                py-immutables
+version             0.9
+categories-append   devel
+platforms           darwin
+supported_archs     noarch
+license             Apache-2
+
+python.versions     35 36 37
+
+maintainers         @jandemter {demter.de:jan} openmaintainer
```
Comment (mf2k) [inline python/py-immutables/Portfile]: See above. This needs to be fixed.

File: python/py-outcome/Portfile [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+name                py-outcome
+version             1.0.0
+categories-append   devel
+platforms           darwin
+supported_archs     noarch
+license             {Apache-2 MIT}
+
+python.versions     27 35 36 37
+
+maintainers         @jandemter {demter.de:jan} openmaintainer
```
Comment (mf2k) [inline python/py-outcome/Portfile]: See above. This needs to be fixed.

File: python/py-outcome/Portfile [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+
+distname            outcome-${version}
+
+checksums           rmd160  6b31076292a1a6ef4ee93bffb0d3cc109a63d515 \
+                    sha256  9d58c05db36a900ce60c6da0167d76e28869f64b338d60fa3a61841cfa54ac71 \
+                    size    17307
+
+if {${name} ne ${subport}} {
+    depends_build-append    port:py${python.version}-setuptools
+    depends_lib-append      port:py${python.version}-attrs
```
Comment (mf2k) [inline python/py-outcome/Portfile]: Given that `noarch` is set, shouldn't this be a `depends_run` instead?

File: python/py-sniffio/Portfile [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+name                py-sniffio
+version             1.1.0
+categories-append   devel
+platforms           darwin
+supported_archs     noarch
+license             {Apache-2 MIT}
+
+python.versions     36 37
+
+maintainers         @jandemter {demter.de:jan} openmaintainer
```
Comment (mf2k) [inline python/py-sniffio/Portfile]: See above. This needs to be fixed.

File: python/py-trio/Portfile [hunk 14]
```diff
@@ -0,0 +1,62 @@
...
+patchfiles          patch-trio-_socket.py-select-fix.diff
+
+if {${name} ne ${subport}} {
+    depends_build-append    port:py${python.version}-setuptools
+    depends_lib-append      port:py${python.version}-sortedcontainers \
+                            port:py${python.version}-async_generator \
+                            port:py${python.version}-attrs \
+                            port:py${python.version}-idna \
+                            port:py${python.version}-outcome \
+                            port:py${python.version}-sniffio
```
Comment (mf2k) [inline python/py-trio/Portfile]: Given that `noarch` is set, shouldn't these dependences be `depends_run` instead?

File: python/py-trio/Portfile [hunk 15]
```diff
@@ -0,0 +1,62 @@
...
+name                py-trio
+version             0.12.1
+categories-append   devel
+platforms           darwin
+supported_archs     noarch
+license             {Apache-2 MIT}
+
+python.versions     36 37
+
+maintainers         @jandemter {demter.de:jan} openmaintainer
```
Comment (mf2k) [inline python/py-trio/Portfile]: See above. This needs to be fixed.

File: python/py-trustme/Portfile [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+name                py-trustme
+version             0.5.2
+categories-append   devel
+platforms           darwin
+supported_archs     noarch
+license             {Apache-2 MIT}
+
+python.versions     27 35 36 37
+
+maintainers         @jandemter {demter.de:jan} openmaintainer
```
Comment (mf2k) [inline python/py-trustme/Portfile]: See above. This needs to be fixed.

File: python/py-trustme/Portfile [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+distname            trustme-${version}
+
+checksums           rmd160  1c5bc086145a8c0c51f2aa71250e5ff25412314f \
+                    sha256  8b804c55c7bcb5186f1f408c9da1e5fda915e6fe0142f4411ea900c380456e80 \
+                    size    21819
+
+if {${name} ne ${subport}} {
+    depends_build-append    port:py${python.version}-setuptools
+    depends_lib-append      port:py${python.version}-cryptography \
+                            port:py${python.version}-idna
```
Comment (mf2k) [inline python/py-trustme/Portfile]: Given that `noarch` is set, shouldn't these dependences be `depends_run` instead?

**Comment by mf2k** [comment]
You will need to change the commit message to conform to our guidelines. 

I think it is better to have one commit per new port. 

`py-trio: New port
`
`py-sniffio: New port
`
etc.

**Comment by reneeotten** [comment]
@jandemter it looks like my initial comments were all addresses, so thank you for that! I am traveling right now and don't have the time to take a closer look, but it looks fine to me - @mf2k can you finalize/merge this please?

**Pull request merged**